funnel/models/proposal.py

```diff
@@ -273,6 +345,17 @@ def permissions(self, user, inherited=None):
                     perms.add('decline-proposal')  # Decline speaking
         return perms
 
+    # Roles
+
+    def roles_for(self, actor=None, anchors=()):
+        roles = super(Proposal, self).roles_for(actor, anchors)
+        if self.speaker and self.speaker == actor:
+            roles.add('speaker')
+        if self.user == actor:
```

Comment:
Make this `if self.user and self.user == actor`. While there is no condition under which `user` will be None _currently_, don't be caught by surprise when it comes up in future.

We made this mistake in Lastuser some years ago: https://github.com/hasgeek/lastuser/commit/17aaa61ff0b06531dc1bd17b7726ad0b680cbcad